a8c-files.php

```diff
@@ -776,6 +776,232 @@ function a8c_files_init() {
+			return $image_meta;
+		}
+
+		// Need the mime type but without getting the attachment post (fast).
+		if ( ! preg_match( '/\.(jpe?g|png|gif)$/i', $image_basename, $extension ) ) {
+			// Something doesn't add up...
+			return $image_meta;
+		}
+
+		$extension = strtolower( $extension[1] );
+
+		if ( 'jpg' === $extension || 'jpeg' === $extension ) {
+			$mime_type = 'image/jpeg';
+		} else {
+			$mime_type = "image/{$extension}";
+		}
+
+		$crop = false;
+
+		// Add the Photon query args from the img src.
+		// Also, when adding the (new) sizes, one of the 'file' values has to match the image src basename
+		// including the query args. Otherwise wp_calculate_image_srcset() bails out and returns false.
+		if ( strpos( $image_src, '?' ) ) {
+			$query = explode( '?', $image_src );
+			wp_parse_str( str_replace( array( '&#38;', '&#038;', '&amp;' ), '&', $query[1] ), $photon_args );
+
+			if ( isset( $photon_args['zoom'] ) ) {
+				// This should only be used by direct requests from JS/devicepx.
+				// No good way to generate proper srcset for "zoomed" images.
+				return $image_meta;
+			}
+
+			if ( isset( $photon_args['fit'] ) ) {
+				// When there are width and height from the img tag (that translate into "w" and "h")
+				// the "fit" arg is not needed. If used, it is probably a "special case" or an error.
+				return $image_meta;
+			}
+
+			if ( isset( $photon_args['crop'] ) ) {
+				if ( $photon_args['crop'] === '1' ) {
+					$crop = true;
+				} else {
+					// No support for Photon cropping using pixel values or percents, yet.
+					return $image_meta;
+				}
+			}
+
+			if ( ! empty( $photon_args['resize'] ) ) {
+				// TODO: maybe see if there is a discrepancy between the width/height from the img tag and the values in 'resize'
+				// Assuming the image must have the width/height set in the 'resize' query arg.
+				$resize = explode( ',', $photon_args['resize'] );
+				if ( count( $resize ) === 2 ) {
+					$size_array = array_map( 'intval', $resize );
+				} else {
+					$resize = null;
+				}
+			}
+		}
+
+		list( $image_width, $image_height ) = $size_array;
+
+		if ( empty( $photon_args ) ) {
+			$photon_args = array(
+				'w' => $image_width,
+				'h' => $image_height,
+			);
+		}
+
+		$sizes = $widths_added = array();
+		$_sizes = self::get_thumbnail_sizes();
+
+		$full_width = intval( $image_meta['width'] );
+		$full_height = intval( $image_meta['height'] );
+
+		if ( ! empty( $resize ) || $crop ) {
+			list( $full_width, $full_height ) = wp_expand_dimensions( $image_width, $image_height, $full_width, $full_height );
+		}
+
+		$width_2x = round( $image_width * 2 );
+		$height_2x = round( $image_height * 2 );
+
+		// Add the current size and 2x current size. That should be the largest available size.
+		// The current "retina" browsers don't show images larger than 2x even on displays with 3.5x or 4.0x pixel density.
+		// Prepend wpcom-current. Fixes iOS8 bug.
```

Comment:
Is there more to this? I don't see any other references to this, so I wonder how and what it fixes.

Reply:
It's related to this ticket: https://core.trac.wordpress.org/ticket/35030 . Similar patch to the one in here can be found in core itself.